Google.Api.Gax.Grpc/RetrySettings.cs

```diff
@@ -179,6 +181,24 @@ public TimeSpan NextBackoff(TimeSpan currentBackoff)
             }
         }
 
+        /// <summary>
+        /// Returns a sequence of already-jittered backoffs. A caller can just iterate over this
+        /// sequence and delay by each returned timespan. This sequence is infinite; callers do not
+        /// need to test the result of MoveNext.
+        /// </summary>
+        /// <param name="initialBackoffOverride">The initial backoff to apply, or null to use <see cref="InitialBackoff"/>;
+        /// this can be used to retry immediately after an initial failure, or to start with a longer-than-normal backoff.</param>
+        /// <returns>A sequence of jittered backoffs, suitable for delays.</returns>
+        public IEnumerator<TimeSpan> GetJitteredBackoffSequence(TimeSpan? initialBackoffOverride = null)
```

Comment:
Why not make this an `IEnumerable<TimeSpan>`, then the code in `WithRetry()` can use a `foreach` loop over it, which would be nice.

Reply:
We *could* do that, I suppose. It would be slightly odd to have "the backoff that I'll try if it fails" in scope the whole time. Will add a second commit to this PR to show what it would look like.

Reply:
+1

Reply:
I've pushed the new commit now - PTAL. If we don't mind "simple rethrow of an exception" rather than "rethrow preserving stack trace", we probably *could* bake MaxAttempts in there...